mm2src/coins/eth.rs

```diff
@@ -1726,114 +1733,75 @@ impl MarketCoinOps for EthCoin {
-                            loop {
-                                if status.ms2deadline().unwrap() < 0 {
-                                    status.append(" Timed out.");
-                                    return ERR!(
-                                        "Waited too long until {} for transaction {:?} confirmation ",
-                                        wait_until,
-                                        tx
-                                    );
-                                }
-
-                                // Make sure that the transaction is returned by eth_getTransactionByHash too so that swaps don't fail at payment validation
-                                // https://github.com/KomodoPlatform/atomicDEX-API/issues/1630#issuecomment-1401736168
-                                match selfi.web3.eth().transaction(TransactionId::Hash(tx.hash)).await {
-                                    Ok(Some(_)) => {
-                                        status.append(" Confirmed.");
-                                        return Ok(());
-                                    },
-                                    Ok(None) => error!(
-                                        "Didn't find tx: {:02x} for coin: {} on RPC node using eth_getTransactionByHash. Retrying in {} seconds",
-                                        tx.hash,
-                                        selfi.ticker(),
-                                        check_every
-                                    ),
-                                    Err(e) => error!(
-                                        "Error {} calling eth_getTransactionByHash for coin: {}, tx: {:02x}. Retrying in {} seconds",
-                                        e,
-                                        selfi.ticker(),
-                                        tx.hash,
-                                        check_every
-                                    ),
-                                }
-
-                                Timer::sleep(check_every as f64).await;
-                            }
+                // Wait for a block that achieves the required confirmations
+                let confirmation_block_number = confirmed_at + required_confirms;
+                if let Err(e) = selfi
+                    .wait_for_block(confirmation_block_number, input.wait_until, check_every)
+                    .compat()
+                    .await
+                {
+                    update_status!(e, status);
+                    return Err(e.to_string());
+                }
+
+                // Make sure that there was no chain reorganization that led to transaction confirmation block to be changed
+                match selfi
+                    .transaction_confirmed_at(tx_hash, input.wait_until, check_every)
+                    .compat()
+                    .await
+                {
+                    Ok(conf) => {
+                        if conf == confirmed_at {
```

Comment:
You can add `status.append(" Confirmed.")` here.

Reply:
Thank you for finding this, looks like I missed it in the refactor.

Reply:
Fixed